lib/backend/k8s/resources/node.go

```diff
@@ -160,11 +160,16 @@ func (c *nodeClient) EnsureInitialized() error {
 }
 
 func (c *nodeClient) Watch(ctx context.Context, list model.ListInterface, revision string) (api.WatchInterface, error) {
-	if len(list.(model.ResourceListOptions).Name) != 0 {
-		return nil, fmt.Errorf("cannot watch specific resource instance: %s", list.(model.ResourceListOptions).Name)
+	// Build watch options to pass to k8s.
+	opts := metav1.ListOptions{ResourceVersion: revision, Watch: true}
+	name := list.(model.ResourceListOptions).Name
+	if len(name) != 0 {
+		// We've been asked to watch a specific node resource.
+		log.WithField("name", name).Debug("Watching a single node")
+		opts.FieldSelector = fields.OneTermEqualSelector("metadata.name", name).String()
```

Comment:
hopefully a quick question.  Does the API server handle this field selector explicitly by constructing the correct watch path for etcdv3, or does this actually end up watching everything, but filter out all other nodes at the API server?

Reply:
I don't know the apiserver code super well, but my quick read of it led me to believe that it watches just this one specific object. It seems aware that we've requested a watch on the name of the resource, and has special case code for that.

Even if it internally watched all keys and filtered, that would still be nicer than re-querying every 5 seconds.

Reply:
https://github.com/kubernetes/apiserver/blob/release-1.9/pkg/registry/generic/registry/store.go#L1152-L1177